src/com/owncloud/android/utils/DisplayUtils.java

```diff
@@ -245,6 +258,67 @@ public static Point getScreenSize(Activity caller) {
     }
 
     /**
+     * Determines if user set folder to grid or list view. If folder is not set itself,
+     * it finds a parent that is set (at least root is set).
+     * @param file
+     * @param storageManager
+     * @return
+     */
+    public static boolean isGridView(OCFile file, FileDataStorageManager storageManager){
+        if (file != null) {
+            OCFile fileToTest = file;
+            OCFile parentDir = null;
+            String parentPath = null;
+
+            SharedPreferences setting = MainApp.getAppContext().getSharedPreferences(
+                    "viewMode", Context.MODE_PRIVATE);
+
+            if (setting.contains(fileToTest.getRemoteId())) {
+                return setting.getBoolean(fileToTest.getRemoteId(), false);
+            } else {
+                do {
+                    if (fileToTest.getParentId() != FileDataStorageManager.ROOT_PARENT_ID) {
+                        parentPath = new File(fileToTest.getRemotePath()).getParent();
+                        parentPath = parentPath.endsWith(OCFile.PATH_SEPARATOR) ? parentPath :
+                                parentPath + OCFile.PATH_SEPARATOR;
+                        parentDir = storageManager.getFileByPath(parentPath);
+                    } else {
+                        parentDir = storageManager.getFileByPath(OCFile.ROOT_PATH);
+                    }
+
+                    while (parentDir == null) {
+                        parentPath = new File(parentPath).getParent();
+                        parentPath = parentPath.endsWith(OCFile.PATH_SEPARATOR) ? parentPath :
+                                parentPath + OCFile.PATH_SEPARATOR;
+                        parentDir = storageManager.getFileByPath(parentPath);
+                    }
+                    fileToTest = parentDir;
+                } while (endWhile(parentDir, setting));
+                return setting.getBoolean(fileToTest.getRemoteId(), false);
+            }
+        } else {
+            return false;
+        }
+    }
+
+    private static boolean endWhile(OCFile parentDir, SharedPreferences setting) {
+        if (parentDir.getRemotePath().compareToIgnoreCase(OCFile.ROOT_PATH) == 0) {
+            return false;
+        } else {
+            return !setting.contains(parentDir.getRemoteId());
+        }
+    }
+
+    public static void setViewMode(OCFile file, boolean setGrid){
+        SharedPreferences setting = MainApp.getAppContext().getSharedPreferences(
+                "viewMode", Context.MODE_PRIVATE);
+
+        SharedPreferences.Editor editor = setting.edit();
+        editor.putBoolean(file.getRemoteId(), setGrid);

```

Comment:
The "problem" is that .putBoolean expects a string, so I have to convert it.
But...done ;)